xbmc/CompileInfo.h

```diff
@@ -33,4 +33,5 @@ class CCompileInfo
   static std::vector<std::string> GetAvailableWindowSystems();
   static std::vector<ADDON::RepoInfo> LoadOfficialRepoInfos();
   static std::string GetPythonVersion();
+  static const std::vector<std::string> GetWebserverExtraWhitelist();
```

Comment:
we shouldn't return plain const here. @jfly could you change this to `static std::vector<std::string> GetWebserverExtraWhitelist();` here and in CompileInfo.cpp.in in a follow up pr, please?

Reply:
Sure thing! Teach me a little c++? What's wrong with returning `const` here?

Reply:
i'll try :)
it is not _wrong_ per se but there's just no point in const qualifying when it's returned by value and it may have performance impacts.

- usually the return value is copied anyway e.g. like you do it later `auto kodiExtraWhitelist = CCompileInfo::GetWebserverExtraWhitelist();` so kodiExtraWhitelist is still modifyable.

-  it limits what can be done with the returned object. you can't use it as an lvalue and it prevents you from taking full advantage of move semantics. (not in this particular case but in general)

the first rule of thumb that i was taught by team kodi ;)
hope i explained correctly so if somebody wants to add something please go ahead.

Reply:
this object is only ever intended to be read, so imho const is fine here.

Reply:
Ok, I definitely don't feel qualified to make an executive decision here. Let's continue this conversation over on https://github.com/xbmc/xbmc/pull/21195?